### DISCUSSION TITLE
Trying to create a basic regex for spanish language without success.

### DISCUSSION THREAD

**Discussion by Idomingog** [microsoft/presidio#802] [Q&A] [answered]
I'm trying to create a basic zip code recognition for spanish language, 
I modify the example given, but appears an error that I couldn't solve.  Anyone could give me a hint? 
Thanks in advance.

from presidio_analyzer import AnalyzerEngine, PatternRecognizer, EntityRecognizer, Pattern, RecognizerResult
from presidio_analyzer.recognizer_registry import RecognizerRegistry
from presidio_analyzer.nlp_engine import NlpEngine, SpacyNlpEngine, NlpArtifacts, NlpEngineProvider

from presidio_anonymizer import AnonymizerEngine

# Create configuration containing engine name and models
configuration = {
    "nlp_engine_name": "spacy",
    "models": [{"lang_code": "es", "model_name": "es_core_news_lg"},  
                     {"lang_code": "en", "model_name": "en_core_web_lg"}],
}

# Create NLP engine based on configuration
provider = NlpEngineProvider(nlp_configuration=configuration)
nlp_engine_with_spanish = provider.create_engine()

# Pass the created NLP engine and supported_languages to the AnalyzerEngine
analyzer = AnalyzerEngine(
    nlp_engine=nlp_engine_with_spanish, 
    supported_languages=["en", "es"]
)

# Define the regex pattern
regex = r"(\b\d{5}(?:\-\d{4})?\b)" # very weak regex pattern
cp_pattern = Pattern(name="cp_ 5d", regex=regex, score=0.01)

# Define the recognizer with the defined pattern
cp_recognizer = PatternRecognizer(supported_entity="ES_ZIP_CODE", 
                                  patterns = [cp_pattern],
                                  context= ["Código postal", "código Postal", "CP", "C.P."],
                                  supported_language='es')

registry = RecognizerRegistry()
registry.add_recognizer(cp_recognizer)
analyzer = AnalyzerEngine(registry=registry)

# Test
cp_result = analyzer.analyze(text="Mi CP es 28001.", entities=["ES_ZIP_CODE"], language='es')
print(f"Result:\n {cp_result}")

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_25648/3379088721.py in <module>
     23 
     24 # Test
---> 25 cp_result = analyzer.analyze(text="Mi CP es 28770.",entities=["ES_ZIP_CODE"], language='es')
     26 print(f"Result:\n {cp_result}")

~\anaconda3\envs\Presidio\lib\site-packages\presidio_analyzer\analyzer_engine.py in analyze(self, text, language, entities, correlation_id, score_threshold, return_decision_process, ad_hoc_recognizers)
    161         # run the nlp pipeline over the given text, store the results in
    162         # a NlpArtifacts instance
--> 163         nlp_artifacts = self.nlp_engine.process_text(text, language)
    164 
    165         if self.log_decision_process:

~\anaconda3\envs\Presidio\lib\site-packages\presidio_analyzer\nlp_engine\spacy_nlp_engine.py in process_text(self, text, language)
     41     def process_text(self, text: str, language: str) -> NlpArtifacts:
     42         """Execute the SpaCy NLP pipeline on the given text and language."""
---> 43         doc = self.nlp[language](text)
     44         return self._doc_to_nlp_artifact(doc, language)
     45 

KeyError: 'es'

**Comment by omri374** [comment, marked as answer]
Hi @Idomingog, from a first glance it appears like you create the AnalyzerEngine object twice. Once with the nlp engine and then with the registry.

Try this :

```python
registry = RecognizerRegistry()
registry.add_recognizer(cp_recognizer)

analyzer = AnalyzerEngine(
nlp_engine=nlp_engine_with_spanish, 
registry=registry, 
supported_languages=["en", "es"]
)

```

**Reply by Idomingog** [reply]
Thanks a lot for your help.
